pilot/pkg/networking/core/route/route.go

```diff
@@ -1242,6 +1242,12 @@ func BuildDefaultHTTPInboundRoute(clusterName string, operation string) *route.R
 		// gRPC requests time out like any other requests using timeout or its default.
 		GrpcTimeoutHeaderMax: Notimeout,
 	}
+	out.GetRoute().RetryPolicy = &route.RetryPolicy{
+		RetryOn: "connect-failure,reset-before-request",
```

Comment:
Why do we want to retry `connect-failure`? What scenarios would fail once but succeed on retry?

Reply:
connect-failure is only needed in cases where connection times out because the service is busy - we have seen this happen few times. Increasing connect timeout will help. So we can remove this if you think it is not needed

Reply:
Retry runs the risk of overloading it further though which could be bad perhaps?

It might not be a bad idea, but its not obviously a good idea to me and seems _slightly_ risky to me

Reply:
ok. Will remove